make/RunTests.gmk

```diff
@@ -36,6 +36,24 @@ include FindTests.gmk
 # Parse global control variables
 ################################################################################
 
+HASH:=\#
+
+define IfAppend
+$(if $(strip $1),$(strip $1)$(strip $2),)
+endef
+
+define IfPrepend
+$(if $(strip $1),$(strip $2)$(strip $1),)
+endef
+
+define TestID
+$(subst .java,,$(subst .java$(HASH),,$(suffix $(notdir $1))))
+endef
+
+define HashTestID
```

Comment:
This one too.

Reply:
I removed HashTestID and added the functionality directly into TestID